snappy/snap_local.go

```diff
@@ -174,11 +168,12 @@ func (s *SnapPart) Channel() string {
 
 // Icon returns the path to the icon
 func (s *SnapPart) Icon() string {
-	if s.m.Icon == "" {
+	found, _ := filepath.Glob(filepath.Join(s.basedir, "meta", "icon.*"))
+	if len(found) == 0 {
 		return ""
 	}
 
-	return filepath.Join(s.basedir, s.m.Icon)
+	return found[0]
```

Comment:
Hmm, this might return various different icons depending on unpredictable filesystem ordering. I'd much rather sort the list of found strings here and return a stable result (whatever it is).